### ISSUE TITLE
Dragonflight (10.0.0) breaks Farmit2

### ISSUE THREAD

**Issue by karpana** [wow-addon/FarmIt2#8]
As was expected, and happens with every new expansion, addons have been breaking as a result of the various UI and API changes.

I brute-force edited the addon to make it stop throwing errors.  And it seems to be working.

Disclaimer: I don't use the currency module, so can't speak to whether it is working or not.

Changes I made:
- Farmit2-Core.lua
> - commented out line 308
> - commented out line 310
> - commented out line 311
> - commented out line 312

- Farmit2-Currency.lua
> - commented out line 56
> - commented out line 228
> - commented out line 229
> - commented out line 230

I'm providing a copy of my work as a "FanFix" release, for those that might like to give this a whirl.

https://www.dropbox.com/s/ix87d8v2g2yf27j/FarmIt2_DF_FanFix.zip?dl=0

**Comment by davidcraig** [comment]
Do you know how to raise a pull request? more than happy to merge in the changes :)

**Comment by karpana** [comment]
Only thing I've found that's not working.... "right-clicking" items to activate them.

adding items to bars, including/excluding bank, setting goals all seem to work.

**Comment by karpana** [comment]
> Do you know how to raise a pull request? more than happy to merge in the changes :)

I'm not sure.
my knowledge of github is really weak.  
(even if i use it myself to maintain one addon, I generally don't use pull requests)

**Comment by davidcraig** [comment]
https://opensource.com/article/19/7/create-pull-request-github seems to be a good guide for how to do a PR :)

**Comment by karpana** [comment]
Once the addon was working, I hadn't thought to go looking for more LUA errors... there were another 28...

No better funcitonality than the last update; specifically still can't "use items" with right click...
but the other farming functions on the bars work fine...

providing another FanFix update ... v2 this time...

https://www.dropbox.com/s/uuuhshpfih9umeg/FarmIt2_DF_FanFix_v2.zip?dl=0

**Comment by Mythris29** [comment]
@karpana thank you for your work on this. So far it seems to be working pretty well for me minus the right click functionality (but you already knew that).

**Comment by karpana** [comment]
I've been poking around trying to figure out why the RightClick Functionality isn't working...
and I now understand why...

In my second update (v2), I removed/commented lines 70-75 of the Farmit2_Button.xml file which is these six lines.
```xml
        <Button name="$parent_Item" inherits="SecureActionButtonTemplate" hidden="true">
          <Attributes>
            <Attribute name="type" type="string" value="item" />
          </Attributes>
          <Size x="36" y="36" />
        </Button>
```
These six lines are wholly responsible for "handling" the right-click functionality of "using" an item that is in our farming bars.
I've tried relocating them to other parts of the XML to no benefit (or detriment).

I'm struggling with why/how this is not working... and could really use some assistance in figuring out "why".
In the meantime, I'll look at other addons that do "buttons" and see if there's something I can figure out.

Alas, I don't anticipate this will help.

I'll keep on plugging away at this.

**Comment by karpana** [comment]
I'm at a complete loss right now...
not sure how to make the built in button handling stuff activate the "use action"... 

So very frustrated at the moment...

**Comment by karpana** [comment]
> Do you know how to raise a pull request? more than happy to merge in the changes :)

Pull Request #9 has been submitted for the initial "brute force fixes".

Also raised 3 related issues for independent correction.
1. Fix "Currency" to support changes 10.0.0 made TokenFrame and Bags (#11 )
2. Fix "Settings" to support changes 10.0.0 made to UI and Settings (#10)
3. Fix "Groups/Buttons" to re-enable "right click to use" capability. (#12)

**Comment by karpana** [comment]
Closing this issue as the initial compatibility, albeit with issues, has now been resolved.
Issues #10 #11 and #12 will be used going forward to track the various updates.
